cmd/qat_plugin/README.md

```diff
@@ -1,39 +1,215 @@
+        * [DPDK QAT demo](#dpdk-qat-demo)
+            * [Prerequisites](#prerequisites)
+            * [Build the image](#build-the-image)
+            * [Deploy the pod](#deploy-the-pod)
+            * [Run the test](#run-the-test)
+        * [OpenSSL QAT demo](#openssl-qat-demo)
+* [Checking for hardware](#checking-for-hardware)
+
+# Introduction
+
+This Intel QAT device plugin provides support for Intel QAT devices under Kubernetes.
+Supported Devices include:
+
+- [Intel&reg; Xeon&reg; with Intel&reg; C62X Series Chipset][1]
+- [Intel&reg; Atom&trade; Processor C3000][2]
+- [Intel&reg; Communications Chipset 8925 to 8955 Series][3]
+- [Intel&reg; Communications Chipset 8900 to 8920 Series][4]
+- [Intel&reg; Atom&trade; Processor C2000][5]
+
+> **FIXME** - we need to verify this list:
+> - do se support C2000
+> - do we need to add D15xx devices
+
+The QAT device plugin provides access to QAT hardware accelerated cryptographic and compression features.
+Demonstrations are provided utilising [DPDK](https://doc.dpdk.org/) and [OpenSSL](https://www.openssl.org/).
+
+[Kata Containers](https://katacontainers.io/) QAT integration is documented in the
+[Kata Containers documentation repository][6].
+
+## Modes and Configuration options
+
+The QAT plugin can take a number of command line arguments, summarised in the following table:
+
+| Argument | Meaning |
+|:-------- |:------- |
+| -debug | enable debug output
+| -dpdk-driver string | DPDK Device driver for configuring the QAT device (default "vfio-pci") |
+| -kernel-vf-drivers string | Comma separated VF Device Driver of the QuickAssist Devices in the system. Devices supported: DH895xCC,C62x,C3xxx and D15xx (default "dh895xccvf,c6xxvf,c3xxxvf,d15xxvf") |
+| -max-num-devices int | maximum number of QAT devices to be provided to the QuickAssist device plugin (default 32) |
+| -mode string | plugin mode which can be either dpdk (default) or kernel (default "dpdk") |
+
+The example [DaemonSet YAML](../../deployments/qat_plugin/qat_plugin.yaml) passes a number of these
+arguments, and takes its default values from the
+[QAT default ConfigMap](../../deployments/qat_plugin/qat_plugin_default_configmap.yaml). The following
+table summarises the defaults:
+
+| Argument | Variable | Default setting | Explanation |
+|:-------- |:-------- |:--------------- |:----------- |
+| -dpdk-driver | `$DPDK_DRIVER` | vfio-pci | A more robust and secure choice than the `igb_uio` alternative |
+| -kernel-vf-drivers | `$KERNEL_VF_DRIVERS` | dh895xccvf,c6xxvf,c3xxxvf,d15xxvf | Modify to suit your hardware setup |
+| -max-num-devices | `$MAX_NUM_DEVICES` | 32 | Modify to suit your hardware setup if necessary |
+| -debug | `$DEBUG` | false | Debug is disabled by default |
+
+For more details on the `-dpdk-driver` choice, see 
+[DPDK Linux Driver Guide](http://dpdk.org/doc/guides/linux_gsg/linux_drivers.html).
+
+> **FIXME** - I could not find any information around `kernel-vf-drivers` on the following site:
```

Comment:
`kernel-vf-drivers` refers to QAT virtual function kernel drivers. They're not defined by DPDK.

Reply:
thx. I think the original doc was wrong. I've pointed to the same kernel link provided in one of your other comments.